From 20c2636a05df8d912dff8a180330593b55edeea1 Mon Sep 17 00:00:00 2001
From: Puunithaaraj Gopal <puunithaaraj.gopal@intel.com>
Date: Wed, 26 Jun 2019 21:54:43 +0800
Subject: [PATCH 1139/1225] media: ipu: Move ipu_buttress_tsc_ticks_to_ns out
 of CONFIG_DEBUG_FS ifdef guard.

Fix compilation error when CONFIG_DEBUG_FS is not enabled.

Tracked-on: OOLI2-3262
Tracked-On: PKT-2253
Signed-off-by: Puunithaaraj Gopal <puunithaaraj.gopal@intel.com>
Change-Id: I683bc926e57b990eace51bb4c3dacaddf7e101f8
---
 drivers/media/pci/intel/ipu-buttress.c | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-buttress.c b/drivers/media/pci/intel/ipu-buttress.c
index 33a6449..7967fe5 100644
--- a/drivers/media/pci/intel/ipu-buttress.c
+++ b/drivers/media/pci/intel/ipu-buttress.c
@@ -1546,22 +1546,6 @@ static int ipu_buttress_start_tsc_sync_set(void *data, u64 val)
 DEFINE_SIMPLE_ATTRIBUTE(ipu_buttress_start_tsc_sync_fops, NULL,
 			ipu_buttress_start_tsc_sync_set, "%llu\n");
 
-u64 ipu_buttress_tsc_ticks_to_ns(u64 ticks)
-{
-	u64 ns = ticks * 10000;
-	/*
-	 * TSC clock frequency is 19.2MHz,
-	 * converting TSC tick count to ns is calculated by:
-	 * ns = ticks * 1000 000 000 / 19.2Mhz
-	 *    = ticks * 1000 000 000 / 19200000Hz
-	 *    = ticks * 10000 / 192 ns
-	 */
-	do_div(ns, 192);
-
-	return ns;
-}
-EXPORT_SYMBOL_GPL(ipu_buttress_tsc_ticks_to_ns);
-
 static int ipu_buttress_tsc_get(void *data, u64 *val)
 {
 	return ipu_buttress_tsc_read(data, val);
@@ -1665,6 +1649,22 @@ int ipu_buttress_debugfs_init(struct ipu_device *isp)
 
 #endif /* CONFIG_DEBUG_FS */
 
+u64 ipu_buttress_tsc_ticks_to_ns(u64 ticks)
+{
+	u64 ns = ticks * 10000;
+	/*
+	 * TSC clock frequency is 19.2MHz,
+	 * converting TSC tick count to ns is calculated by:
+	 * ns = ticks * 1000 000 000 / 19.2Mhz
+	 *    = ticks * 1000 000 000 / 19200000Hz
+	 *    = ticks * 10000 / 192 ns
+	 */
+	do_div(ns, 192);
+
+	return ns;
+}
+EXPORT_SYMBOL_GPL(ipu_buttress_tsc_ticks_to_ns);
+
 static ssize_t
 ipu_buttress_psys_fused_min_freq_get(struct device *dev,
 				     struct device_attribute *attr, char *buf)
-- 
2.7.4

